### DISCUSSION TITLE
Enable prompt caching for AWS Bedrock Sonnet 3.5 V2

### DISCUSSION THREAD

**Discussion by ppeach** [cline/cline#897] [Feature Requests]
Discord chat here

https://discord.com/channels/1275535550845292637/1315410865813585938

it's probably blocked by
> Support prompt caching in bedrock API (@anthropic-ai/bedrock-sdk) #507
https://github.com/anthropics/anthropic-sdk-typescript/issues/507

after that is done by anthropic

a subsequent change is probably needed in cline's `package.json`
> `    "dependencies": {`
> `        "@anthropic-ai/bedrock-sdk": "^0.10.2",`
https://github.com/cline/cline/blob/main/package.json#L154-L155
to use a newer version of anthropic's `bedrock-sdk` that fixes https://github.com/anthropics/anthropic-sdk-typescript/issues/507

followed by a new `cline/cline` release
https://github.com/cline/cline/releases
for publishing to
https://marketplace.visualstudio.com/items?itemName=saoudrizwan.claude-dev

**Comment by lloydchang** [comment]
As the original author who wrote the above snippet in Discord, I'm in support of the feature request. Thanks!

**Comment by shrijayan** [comment]
https://github.com/cline/cline/discussions/1208

In AWS Bedrock, they support prompt caching through Prompt Routers, which have implemented caching for Anthropic and Meta models. They provide the model ARN ID to access the models. By implementing **Accept Model ARN ID in AWS Bedrock Model Selection #1208**, where we can input the model ARN ID in AWS Bedrock, this feature will be highly useful.
